test-runner/src/main/scala/com/typesafe/spark/test/mesos/RolesSpec.scala

```diff
@@ -23,16 +43,23 @@ trait RolesSpec { self: MesosIntTestHelper =>
     assert(m.sparkFramework.isDefined, "The driver should be running")
 
     if (cfg.role != "*") {
-      // TODO: add message
-      assert(m.slaves.flatMap { x => x.roleResources.map { y => y.roleName } }.contains(cfg.role))
+      assert(m.slaves.flatMap { x => x.roleResources.map { y => y.roleName } }.contains(cfg.role), "spark role should exist")
 
-      // TODO: add message
-      assert(m.sparkFramework.get.resources.cpu == cfg.roleCpus.toInt)
+      // make sure reserved resources for that role are used
+      m.slaves.foreach {
+        x =>
+          val reserved = x.roleResources.filter(r => r.roleName == cfg.role).head
+          assert(reserved.resources.cpu == cfg.roleCpus.toInt, "Reserved resources per slave")
```

Comment:
This doesn't test much. It basically check that the value in `cfg.roleCpus.toInt` matches the configuration of the Mesos slave. `reserved.resources.cpu` is immutable during the tests.